src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -662,8 +698,16 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
 
 	private restoreViewState(): void {
 		if (this.savedViewState) {
-			this.slickgrids.toArray()[0].selection = this.savedViewState.gridSelections;
-			let viewport = ((this.slickgrids.toArray()[0] as any)._grid.getCanvasNode() as HTMLElement).parentElement;
+			//get all rows shown in gridSelections Range to use setSelectedRows directly.
+			let rowArray: number[] = [];
+			let fromRow = this.savedViewState.gridSelections[0].fromRow;
+			let toRow = this.savedViewState.gridSelections[0].toRow;
+			for (let i = fromRow; i <= toRow; i++) {
+				rowArray.push(i);
+			}
+
+			this._tables[0].setSelectedRows(rowArray);
```

Comment:
Do we know why tables is an array if we're special-casing item `0` so frequently?  Maybe we should just have a `table` item?